sdk/inc/azure/iot/az_iot_pnp_client.h

```diff
@@ -650,21 +650,22 @@ AZ_NODISCARD az_result az_iot_pnp_client_twin_end_property_with_status(
  * @brief Read the IoT Plug and Play twin properties version.
  *
  * @param[in] client The #az_iot_pnp_client to use for this call.
- * @param[in,out] ref_json_reader The #az_json_reader used to parse through the JSON payload.
+ * @param[in] json_reader The #az_json_reader used to parse through the JSON payload. An internal
+ * copy is made to maintain the index of \p json_reader.
  * @param[in] response_type The #az_iot_pnp_client_twin_response_type representing the message type
  * associated with the payload.
  * @param[out] out_version The numeric version of the properties in the JSON payload.
  *
  * @pre \p client must not be `NULL`.
- * @pre \p ref_json_reader must not be `NULL`.
+ * @pre \p json_reader must not be `NULL`.
  * @pre \p out_version must not be `NULL`.
  *
  * @return An #az_result value indicating the result of the operation.
  * @retval #AZ_OK If the function returned a valid version.
  */
 AZ_NODISCARD az_result az_iot_pnp_client_twin_get_property_version(
     az_iot_pnp_client const* client,
-    az_json_reader* ref_json_reader,
+    az_json_reader* json_reader,
```

Comment:
In that case, why not just accept the reader by value and you get the copy for free?
```suggestion
    az_json_reader json_reader_copy,
```

Reply:
Yea I thought about it. I can make the change.

Reply:
No, we don't. I just thought the hint would be nice, but I think either name will be fine (or even a different one).

The documentation should convey: The state of the reader won't change. Giving some indication in the API signature (other than the fact that we don't take a pointer), might make the  code more self-documenting.